### ISSUE TITLE
Zero Hydra Failures for 16.09

### ISSUE THREAD

**Issue by domenkozar** [NixOS/nixpkgs#18209]
This is a call for testing of [release-16.09](https://github.com/NixOS/nixpkgs/tree/release-16.09) branch and fixing build failures.

Build failures of latest evaluation on http://hydra.nixos.org/jobset/nixos/release-16.09:

---

- [ ] [nixos.tests.grsecurity.i686-linux](http://hydra.nixos.org/build/40684687)
  - maintainers: @copumpkin, @joachifm
- [x] [nixos.tests.mathics.i686-linux](http://hydra.nixos.org/build/40674177)
  - maintainers: @benley
- [x] [nixos.tests.mathics.x86_64-linux](http://hydra.nixos.org/build/40669814)
  - maintainers: @benley
- [x] [nixos.tests.networking.networkd.macvlan.i686-linux](http://hydra.nixos.org/build/40687294)
  - maintainers: @wkennington
- [x] [nixos.tests.networking.networkd.macvlan.x86_64-linux](http://hydra.nixos.org/build/40690193)
  - maintainers: @wkennington
- [ ] [nixos.tests.pumpio.i686-linux](http://hydra.nixos.org/build/40683104)
  - maintainers: @rvl
- [ ] [nixos.tests.virtualbox.net-hostonlyif](http://hydra.nixos.org/build/40690563)
  - maintainers: @aszlig, @wkennington
- [ ] [nixpkgs.amdappsdkFull.i686-linux](http://hydra.nixos.org/build/40674388)
  - maintainers: @offlinehacker 
- [ ] [nixpkgs.amdappsdkFull.x86_64-linux](http://hydra.nixos.org/build/40693335)
  - maintainers: @offlinehacker 
- [ ] [nixpkgs.linuxPackages_chromiumos_3_14.ena.x86_64-linux](http://hydra.nixos.org/build/40678486)
  - maintainers: @edolstra 
- [ ] [nixpkgs.linuxPackages_chromiumos_3_14.virtualboxGuestAdditions.x86_64-linux](http://hydra.nixos.org/build/40674923)
  - maintainers: @sander
- [x] [nixpkgs.linuxPackages_chromiumos_3_18.lttng-modules.x86_64-linux](http://hydra.nixos.org/build/40669306)
  - maintainers: @bjornfor
- [x] [nixpkgs.linuxPackages_chromiumos_latest.lttng-modules.x86_64-linux](http://hydra.nixos.org/build/40692727)
  - maintainers: @bjornfor
- [ ] [nixpkgs.linuxPackages_grsec_nixos.virtualboxGuestAdditions.i686-linux](http://hydra.nixos.org/build/40685836)
  - maintainers: @sander
- [ ] [nixpkgs.linuxPackages_grsec_nixos.virtualboxGuestAdditions.x86_64-linux](http://hydra.nixos.org/build/40670574)
  - maintainers: @sander
- [x] [nixpkgs.mysqlWorkbench.x86_64-linux](http://hydra.nixos.org/build/40671281)
  - maintainers: Karn
- [x] [nixpkgs.x42-plugins.x86_64-linux](http://hydra.nixos.org/build/40671510)
  - maintainers: @magnetophon

cc @NixOS/nixpkgs-committers

**Comment by dezgeg** [comment]
`go-repo-root` is fallout from my changes. I will tackle that.

**Comment by shlevy** [comment]
Fixed phc-intel stuff in https://github.com/NixOS/nixpkgs/commit/b3871e4b7bd64fd82f5353ad1e70c773eddfd42c

**Comment by kamilchm** [comment]
It looks like Mesos errors are caused by `readdir_r` deprecation in `libc` -> https://sourceware.org/ml/libc-alpha/2016-02/msg00093.html
Possible resolutions:
1. Ignore deprecation warning in Mesos build
2. Prepare a patch in `nixpkgs` and submit it to the upstream
3. Use older `libc`

**Comment by dezgeg** [comment]
Can anyone locally reproduce these: `pkgsi686Linux.librem pkgsi686Linux.sysklogd sysklogd`? They all work for me.

**Comment by zimbatm** [comment]
^ got a patch for packer if interested

**Comment by kamilchm** [comment]
^ Mesos patches backported from 1.0

It may be worth to include latest bugfixes for Mesos and Marathon in 16.09:
* https://github.com/NixOS/nixpkgs/pull/18220
* https://github.com/NixOS/nixpkgs/pull/18221

**Comment by garbas** [comment]
for ``nixpkgs.linuxPackages_4_7.spl.*`` fixes are in [their master](https://github.com/zfsonlinux/spl/commit/fdbc1ba99d1f4d3958189079eee9b6c957e0264b) but no idea if they plan any release soon.

**Comment by shlevy** [comment]
@garbas Does the patch apply cleanly? If not IMO we should  mark spl broken on >= 4.7 until the release

**Comment by obadz** [comment]
Fixed DBIxClass/hydra

**Comment by obadz** [comment]
@dezgeg: sysklogd build fine here too.

**Comment by shlevy** [comment]
I'll restart sysklogd build

**Comment by obadz** [comment]
@rushmorem: 940dc9b didn't seem to fix lizardfs

**Comment by rushmorem** [comment]
@obadz I'm still at work right now. Will work on it as soon as I'm free.

**Comment by obadz** [comment]
Fixed netatalk

**Comment by rushmorem** [comment]
`lizardfs` should be fixed now. I did not have `nix.useSandbox` enabled in my `/etc/nixos/configuration.nix`.

**Comment by shlevy** [comment]
Looks like there was a significant rebuild in the latest eval, I cancelled non-current builds to give hydra some breathing room.

**Comment by obadz** [comment]
Looks like `minc_tools` tests have never passed. cc @bcdarwin original contributor via ef8e0d6.

**Comment by domenkozar** [comment]
I've updated the issue with the new failing builds.

**Comment by rycee** [comment]
Fixed `nixpkgs.trigger.i686-linux` and `nixpkgs.trigger.x86_64-linux` in 8e3ea885e09a261b7d54c8a609f535e4dfc18290.

**Comment by rycee** [comment]
Also did `nixpkgs.teeworlds` in 69ac966f95ad117a263ef9084b5cad1c3cfccccc.

**Comment by lancelotsix** [comment]
I have been to build `pypy` just fine locally from the release-16.09 branch, can someone try to relaunch the hydra build?

```
$ nix-build -A pypy
/nix/store/kakm6a50q46907w32ij4z3bda303mnh3-pypy-5.3.1
$ /nix/store/kakm6a50q46907w32ij4z3bda303mnh3-pypy-5.3.1/bin/pypy --version
Python 2.7.10 (7e8df3df96417c16c2d55b41352ec82c9c69c978, Sep 03 2016, 07:01:29)
[PyPy 5.3.1 with GCC 5.4.0]
```

**Comment by vcunat** [comment]
Restarted now, but over the long term, it would be nice if the tests were more reliable. Pypy tests tend to break relatively often, and that's no good as it takes so long to build.

**Comment by garbas** [comment]
@lancelotsix @vcunat do we know which tests from pypy are not reliable? i think this would be worth reporting upstream.

**Comment by FRidh** [comment]
just updated pypy and julia and both built

**Comment by vcunat** [comment]
x42-plugins has been repeatedly succeeding and failing for a long time, even on 16.03 http://hydra.nixos.org/job/nixos/release-16.03/nixpkgs.x42-plugins.i686-linux/all I never managed to find out how that could happen.

**Comment by garbas** [comment]
@domenkozar spl on kernels >= 4.7 are now marked as broken until new spl release is there. you want us to check off stuff from the list or you want to do it?

**Comment by vcunat** [comment]
I was check-boxing myself on that list, for stuff I fixed or verified to build.

**Comment by joachifm** [comment]
All the dpdk errors are the same. There's an update available, testing the build now.

**Comment by FRidh** [comment]
i686 version of julia still fails.
```
line 118: ErrorException("error compiling primes: could not load library "libopenlibm"
```
even though it is included.

**Comment by obadz** [comment]
`nixpkgs.tarball` broken by 5ce6cd0 cc @svanderburg

**Comment by domenkozar** [comment]
I've updated the issue with latest failing builds from http://hydra.nixos.org/eval/1290819

There is a new evaluation pending, which should fix some tests.

cc @peti for haskell packages.

**Comment by domenkozar** [comment]
Fixed tarball in ad771764743d9645084ac21730bb5a6cc11ce672 /cc @fpletz

**Comment by domenkozar** [comment]
Last failing dependency of `tested` job is OVA, I'm suspecting the following change: https://github.com/NixOS/nixpkgs/pull/17822#issuecomment-244598190

**Comment by Mic92** [comment]
Regarding `nixos.tests.ferm.i686-linux`:

This is only broken on 32bit not x86_64: Is there a way to build a 32bit integration test on a 64bit installation?

**Comment by Mic92** [comment]
Is there a way to automatically inform all maintainers of these packages? I only saw this by a accident.

**Comment by domenkozar** [comment]
@Mic92 I'm working on a script to figure out when the build(s) started failing and who to really ping.

**Comment by domenkozar** [comment]
@Mic92 you can run it like `$ nix-build nixos/release.nix -A tests.ferm.i686-linux`

**Comment by vcunat** [comment]
@domenkozar: doesn't Hydra do that with `meta.maintainers` already?

**Comment by domenkozar** [comment]
@vcunat if we had a strict rule that each module/pkgs would have `meta.maintainers`, that would work, yes. Otherwise I usually have to pin down the commit, just like mention bot does.

**Comment by vcunat** [comment]
Forcing people wouldn't help much, I think.

**Comment by domenkozar** [comment]
I agree, I'm just saying that `meta.maintainers` is partially useful here.

**Comment by Mic92** [comment]
ok, maybe I was not informed about this breakage by hydra, because I added the integration test after submitting the package.

**Comment by aristidb** [comment]
ispc should be fixed now. It's merged to 16.09, but not yet evaluated by Hydra.

**Comment by Mic92** [comment]
`nixos.tests.ferm.i686-linux` will be fixed with: https://github.com/NixOS/nixpkgs/pull/18288

**Comment by lancelotsix** [comment]
`nixpkgs.salut_a_toi` fixed in c47eb4a43a9ab338c26b6b05d59ad0cb90507ed1

**Comment by domenkozar** [comment]
I've opened a separate issue to address slow `webkitgtk24` build: https://github.com/NixOS/nixpkgs/issues/18312

**Comment by obadz** [comment]
gtk-sharp* no longer builds, probably due to staging merge 8c4aeb1, http://hydra.nixos.org/build/39585120/nixlog/1#line-2564 cc @dezgeg

**Comment by lancelotsix** [comment]
`nixpkgs.netcdf-mpi` fixed in 396ffac28f948ebe4864551c67a01d1828e22ba6

**Comment by domenkozar** [comment]
I've updated the failing builds, they now include maintainers.

**Comment by shlevy** [comment]
@domenkozar Poor @eelco getting saddled with @edolstra's packages!

**Comment by domenkozar** [comment]
I guess we should first make sure people comply with out guidelines in `lib/maintainers.nix`.

**Comment by obadz** [comment]
`minc_tools` fixed by disabling tests. github won't let me tick the checkbox in the original post ✓

**Comment by obadz** [comment]
d605380 is going to be the first channel release: http://hydra.nixos.org/build/39927851#tabs-constituents

**Comment by kampfschlaefer** [comment]
I will look at the container tests. They seem to fail for strange flaky reasons.

**Comment by pxc** [comment]
@domenkozar Think you've got the wrong maintainer for the Gazebo stuff. Nothing to do with me (@pxc).

**Comment by kampfschlaefer** [comment]
@domenkozar I think I know now why the container tests fail irregularly: It seems to be with the merge 509733a34393b5cb15dbb04ed597e069e08d345a (*) where the filesystem handling in stage1 and stage2 was changed. From what I gather, with the merge itself, even VMs like from qemu stopped working, but that seems to got fixed rather soon. But within the nixos containers its still broken.

Sadly my knowledge of these internals is not (yet) good enough to dive in and fix it in a timely manner. Maybe someone with more knowledge can take a look at it?


(*) cb4cd5d7f03a12cc53b1ce5bd83572786a5f6ab7 before that builds the container tests just fine.

**Comment by domenkozar** [comment]
cc @abbradar could you address https://github.com/NixOS/nixpkgs/issues/18209#issuecomment-244819034? Thanks!

**Comment by abbradar** [comment]
I'll be on it this evening GMT+3 (in ~8 hours).

**Comment by groxxda** [comment]
`fstar` is still broken :disappointed:

**Comment by obadz** [comment]
@groxxda, `fstar` builds fine locally.

**Comment by groxxda** [comment]
@obadz for me too but not on hydra. should be disable checks? http://hydra.nixos.org/build/39996818

**Comment by obadz** [comment]
I think it would be better to report the issue upstream?

**Comment by obadz** [comment]
And maybe open an issue in nixpkgs as well.

**Comment by kampfschlaefer** [comment]
I started a pull request for the container tests: #18377 
Currently it would make the tests green reliable. But the containers will still wait for a minute at startup for dev-vda. And I have no clue yet why.

**Comment by abbradar** [comment]
Hopefully containers are fixed in https://github.com/NixOS/nixpkgs/commit/aed2cd32f8069ae114a82b25648c5e67c060ae39.

**Comment by obadz** [comment]
`seq24` fixed via d0bf785

**Comment by obadz** [comment]
`haskellPackages.haste-Cabal` marked broken in 0271f37

**Comment by groxxda** [comment]
Please backport #18334 to fix linphone. 
Somehow the `i686` build is broken again on master (after building successfully once) due to a dependency (`belle-sip`) not building anymore, not sure if release 16.09 is also affected :disappointed:

**Comment by 7c6f434c** [comment]
>Somehow the `i686` build is broken again on master (after building successfully once) due to a dependency (`belle-sip`) not building anymore, not sure if release 16.09 is also affected :disappointed: 

I think the situation on `master` is the following: after a small update
just to fix build I managed to finish a larger update of linphone with
all its dependencies to the true latest version.

Fresh `linphone` does build locally for me on `i686` and `x86_64`, so  
I think that Hydra has a transient failure.

**Comment by domenkozar** [comment]
Done.

I've again updated the failing builds, this time without maintainers (since that doesn't work well yet).

Notes:

- virtualbox will get fixed with https://github.com/NixOS/nixpkgs/pull/17502/
- @peti added lots of Haskell packages and he'll update the failing builds once they all finish
- I'll fix `nova` and `keystone`

@therealpxc would you fix Gazebo?
@ttuegel would you fix a few QT failures?

That should leave us with just a few dozen builds and tests.

**Comment by groxxda** [comment]
@7c6f434c oh i didnt see you updated linphone to the newest version :+1: 
We probably want the newer linphone backported when it works on master for i686 since my version lacks LIME support (didn't realize we have mbedtls_1_3)

I'm currently building the gazebo attrset and will report back when it finished

**Comment by 7c6f434c** [comment]
Separate `xulrunner` seems to be dead by now, and `firefox --app` seems to work — should we remove `xulrunner` package or make a compatibility wrapper?

**Comment by domenkozar** [comment]
I'd just remove it, waiting on confirmation from @edolstra sincehe's the maintainer

**Comment by 7c6f434c** [comment]
`fstar` and `linphone` build fine on `master` right now… (and I haven't touched anything related today)

**Comment by joachifm** [comment]
I think the docker registry test can be disabled. The service module is based on software that has long been deprecated and the test itself has been failing since before the previous release (iirc).

**Comment by domenkozar** [comment]
@offlinehacker are you willing to fix docker-registry test?

**Comment by domenkozar** [comment]
Updated. We're getting close :)

**Comment by domenkozar** [comment]
@dezgeg are you still planning to fix go-repo-root?

**Comment by obadz** [comment]
`vrpn` fixed in 588e17b

**Comment by obadz** [comment]
the `x42-plugins` error is really strange: looks like the tar.xz file can't be unpacked, but the same file with the same hash works fine in other builds…

http://hydra.nixos.org/build/39823818/nixlog/1/raw
```
unpacking source archive /nix/store/jg8rqgbprijqg2fmcgd2zj80azg783hm-x42-plugins-20160619.tar.xz
do not know how to unpack source archive /nix/store/jg8rqgbprijqg2fmcgd2zj80azg783hm-x42-plugins-20160619.tar.xz
builder for ‘/nix/store/qbxcpqh6dj551qhg9v5ra2p68dhl4rrh-x42-plugins-20160619.drv’ failed with exit code 1
```

The very same xz file is unpacked just fine here:
http://hydra.nixos.org/build/40037087/log/raw


```
unpacking source archive /nix/store/jg8rqgbprijqg2fmcgd2zj80azg783hm-x42-plugins-20160619.tar.xz
source root is x42-plugins-20160619
```

**Comment by NeQuissimus** [comment]
@obadz I've seen errors like this before when there in an unexpected EOF (happens when my Wifi is flaky).

**Comment by obadz** [comment]
@NeQuissimus, I know the error you're referring to, I don't think that's it. In this case the `.tar.xz` isn't being downloaded, it is already in the nix store.

**Comment by peti** [comment]
The latest evaluation (e3bdf6b) has zero Haskell build and/or evaluation  errors.

**Comment by obadz** [comment]
`!m peti`

**Comment by groxxda** [comment]
Only [haskellPackages.servant-router](http://hydra.nixos.org/build/40065332) timed out during tests.

**Comment by the-kenny** [comment]
`pkgs.mars` should build now.

**Comment by ttuegel** [comment]
I'm investigating the KDE 5 test failure now. The test times out waiting for a window to open. The log is filled with these two lines:
```
machine: must succeed: xwininfo -root -tree | sed 's/.*0x[0-9a-f]* \"\([^\"]*\)\".*/\1/; t; d'
machine: exit status 0
```
Isn't `exit status 0` considered success? Why does this time-out?

**Comment by domenkozar** [comment]
@FRidh any ideas why http://hydra.nixos.org/build/40071046/nixlog/1 fails to build?

**Comment by dezgeg** [comment]
@domenkozar Yes, looking into go-repo-root over the weekend.

**Comment by domenkozar** [comment]
@vcunat @viric gnat45 also fails to build: http://hydra.nixos.org/build/39681681

**Comment by domenkozar** [comment]
@cstrahan would you fix redis-desktop-manager? http://hydra.nixos.org/build/39993538/nixlog/2/raw

**Comment by domenkozar** [comment]
@offlinehacker Could you fix amdappsdkFull?

**Comment by ttuegel** [comment]
The failing KDE 5 test is fixed (and now enabled!) in 701f02a6eed902835e3d2a27838723cd9a7bb66a.

**Comment by domenkozar** [comment]
@wizeman are you willing to talk with fstar upstream about transient failures we've been seeing?

**Comment by offlinehacker** [comment]
@domenkozar docker registry package, nixos module and tests can be happily removed as docker has released https://github.com/docker/distribution and original docker registry it's long time deprecated,.

**Comment by offlinehacker** [comment]
@domenkozar I don't have any amd gpu-s to test amdappsdk, cat it be marked as broken until someone has time to test and fix it?

**Comment by offlinehacker** [comment]
@domenkozar or, let me try to fix it, i will take time this weekend, i might spin up some ethereum miners just for fun ;)

**Comment by cstrahan** [comment]
@domenkozar yeah, I'll try to fix redis-desktop-manager tonight or tomorrow.

**Comment by vcunat** [comment]
We only use gnat for vhdl within nixpkgs, so no high priority IMO, but it seems like some classic mutliple-output problem, as it tries to find `${glibc.out}/include`.

**Comment by vcunat** [comment]
Adelaide's been resurrected again: 36ef583e.

**Comment by obadz** [comment]
@jgeerds, are you trying to fix `tmuxp` ?

**Comment by obadz** [comment]
@jgeerds, if so are you using `nix.useSandbox = true;` ?

**Comment by jgeerds** [comment]
On Sat, Sep 10, 2016, at 02:34, obadz wrote:
> @jgeerds, if so are you using `nix.useSandbox = true;` ?


Yes, I'm trying to fix it :-)

**Comment by jgeerds** [comment]
@obadz You were right, `nix.useSandbox` was disabled on one of my workstations. I fixed `tmuxp` on `master` and `release-16.09`. Sorry for the trouble

**Comment by FRidh** [comment]
I've marked the i686 versions of julia and julia-git as broken.

**Comment by domenkozar** [comment]
Down to 56!

**Comment by domenkozar** [comment]
@puffnfresh would you check why pijul transiently fail on 32bit linux and try to fix that? :)

**Comment by domenkozar** [comment]
@abbradar any ideas about why mumble test fails on 32bit:

```
client2# Plugins: Failed to load libmumble.so.1.2.16: Cannot load library /nix/store/6jkv2vxyybgjw44ygni672a6qx5dzwss-mumble-1.2.16/lib/libmumble.so.1.2.16: (/nix/store/6jkv2vxyybgjw44ygni672a6qx5dzwss-mumble-1.2.16/lib/libmumble.so.1.2.16: undefined symbol: glPopClientAttrib)
```

**Comment by domenkozar** [comment]
Failing tests:

- mathics @benley 
- cadvisor @offlinehacker 
- keyman @aszlig 
- mysql-replication @edolstra 
- tomcat @edolstra

**Comment by domenkozar** [comment]
@exi @HappyEnte @Phreedom kmymoney fails on 32bit build, any objections marking it as broken on i686?

```
[ 73%] Linking CXX executable mymoneyseqaccessmgrtest
CMakeFiles/mymoneyseqaccessmgrtest.dir/mymoneyseqaccessmgrtest.cpp.o: In function `MyMoneySeqAccessMgrTest::testEmptyConstructor()':
mymoneyseqaccessmgrtest.cpp:(.text+0x13cac): undefined reference to `bool QTest::qCompare<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*, char const*, char const*, int)'
mymoneyseqaccessmgrtest.cpp:(.text+0x13cfa): undefined reference to `bool QTest::qCompare<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*, char const*, char const*, int)'
mymoneyseqaccessmgrtest.cpp:(.text+0x13d42): undefined reference to `bool QTest::qCompare<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*, char const*, char const*, int)'
mymoneyseqaccessmgrtest.cpp:(.text+0x13d8a): undefined reference to `bool QTest::qCompare<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*, char const*, char const*, int)'
mymoneyseqaccessmgrtest.cpp:(.text+0x13dd2): undefined reference to `bool QTest::qCompare<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*, char const*, char const*, int)'
CMakeFiles/mymoneyseqaccessmgrtest.dir/mymoneyseqaccessmgrtest.cpp.o:mymoneyseqaccessmgrtest.cpp:(.text+0x13e1a): more undefined references to `bool QTest::qCompare<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*, char const*, char const*, int)' follow
collect2: error: ld returned 1 exit status
make[2]: *** [kmymoney/mymoney/storage/CMakeFiles/mymoneyseqaccessmgrtest.dir/build.make:191: kmymoney/mymoney/storage/mymoneyseqaccessmgrtest] Error 1
make[1]: *** [CMakeFiles/Makefile2:2740: kmymoney/mymoney/storage/CMakeFiles/mymoneyseqaccessmgrtest.dir/all] Error 2
make: *** [Makefile:161: all] Error 2
builder for ‘/nix/store/96ryvhwbh2ss5gq07isj3hsx74kcaf3n-kmymoney-4.8.0.drv’ failed with exit code 2
```

**Comment by exi** [comment]
@domenkozar no objections from my side, I'll try to debug this when i have time.

**Comment by exi** [comment]
@domenkozar Just had a look at this, this needs to be fixed upstream but we can easily include a patch to fix it for us for now. The patch is really simple. Would it be an option to just include a patch on our side for now, while i try to bring it upstream?

**Comment by domenkozar** [comment]
Patches that fix builds are always welcome until upstream fixes the issues.

**Comment by exi** [comment]
@domenkozar, @abbradar The error mentioned in https://github.com/NixOS/nixpkgs/issues/18209#issuecomment-246202704 also happens on 64bit. It's the same as this one: https://bugs.launchpad.net/ubuntu/+source/mumble/+bug/1056907

It does not prevent mumble from launching or working, just causes issues with the mumble game overlay.

It is unrelated to the failed test.

**Comment by benley** [comment]
I can check on mathics later tonight (US east coast)

**Comment by cstrahan** [comment]
I don't know how to fix the Redis Desktop Manager package:

    Could not find qmake configuration file linux-g++.

@ttuegel Do you know what the problem might be? This worked at one point in time.

**Comment by ttuegel** [comment]
@cstrahan You can't override `configureHook` like that. At a minimum, you must call `runHook preConfigure; runHook postConfigure`.

**Comment by cstrahan** [comment]
@ttuegel Oh, the QT setup-hooks register pre/post configure hooks that must run?

Any way, thanks! This seems to be sufficient:

    configurePhase = ''
      runHook preConfigure
      runHook postConfigure
    '';

**Comment by cstrahan** [comment]
PR for RDM here: https://github.com/NixOS/nixpkgs/pull/18543

**Comment by exi** [comment]
I hope the mumble vm test is fixed as of https://github.com/NixOS/nixpkgs/pull/18545

**Comment by domenkozar** [comment]
Here are the final outstanding issues:

- dozen tests fail, see the list at the top (including mumble)
- lttng modules fail on chromiumos kernel (we should probably just mark that as broken)
- fstar needs a maintainer decision based on upstream feedback at https://github.com/FStarLang/FStar/issues/676

Following are being worked on:

- amdappsdkFull is being worked on @offlinehacker 
- go-repo-root is being worked on @dezgeg 
- mysqlWorkbench is being worked on by Karn
- virtualbox https://github.com/NixOS/nixpkgs/pull/18540

**Comment by wmertens** [comment]
the dvorak test seems likely to be an intermittent issue. Can the test be re-run?

**Comment by abbradar** [comment]
I've run `tests.mumble.i686-linux` a few times but can't reproduce the error. This seems strange -- I'm not sure how to debug this.

**Comment by domenkozar** [comment]
@abbradar hydra machines are under I/O load, so usually timings are not set correctly (or better say, somewhere it's missing a statement to wait for a resource to be available)

**Comment by domenkozar** [comment]
@offlinehacker panamax test is also under your maintenance

**Comment by domenkozar** [comment]
All remaining failures now have their corresponding maintainer, except for keymap tests (which I think @aszlig added).

I'll just disable chromiumos lttng builds.

**Comment by offlinehacker** [comment]
@domenkozar can we disable panamax test, and maybe even remove package, because as far as i know noone is still using panamax, i think also centurylinklabs has more or less deprecated it.

**Comment by domenkozar** [comment]
@offlinehacker done in cc841a4841a6ffc13abac26a6de7ffe208a67524, what about other tests?

**Comment by domenkozar** [comment]
Following tests still need heroes :) 

- [ ] [nixos.tests.cadvisor](http://hydra.nixos.org/build/40161962)
  - maintainers: @offlinehacker 
- [ ] [nixos.tests.grsecurity.i686-linux](http://hydra.nixos.org/build/40305832)
  - maintainers: @copumpkin, @joachifm
- [ ] [nixos.tests.keymap.dvorak.x86_64-linux](http://hydra.nixos.org/build/40183285)
- [ ] [nixos.tests.keymap.neo.i686-linux](http://hydra.nixos.org/build/40183319)
- [ ] [nixos.tests.keymap.neo.x86_64-linux](http://hydra.nixos.org/build/40183281)
- [ ] [nixos.tests.mathics.i686-linux](http://hydra.nixos.org/build/40188470)
  - maintainers: @benley
- [ ] [nixos.tests.mathics.x86_64-linux](http://hydra.nixos.org/build/40188454)
  - maintainers: @benley 
- [ ] [nixos.tests.mysqlReplication.i686-linux](http://hydra.nixos.org/build/40162068)
  - maintainers: @edolstra , @chaoflow, @shlevy
- [ ] [nixos.tests.pumpio.i686-linux](http://hydra.nixos.org/build/40183334)
  - maintainers: @rvl
- [ ] [nixos.tests.tomcat.i686-linux](http://hydra.nixos.org/build/40162153)
  - maintainers: @edolstra , @chaoflow

**Comment by shlevy** [comment]
"master# [  105.352192] systemd[1]: mysql.service: Start-pre operation timed out. Terminating." I'm guessing this may be a hydra load issue, trying to restart.

**Comment by exi** [comment]
@domenkozar looking at http://hydra.nixos.org/build/40496245 in the raw log:

```
machine# Unknown partition table signature
machine# [  856.479515] blk_update_request: I/O error, dev fd0, sector 0
machine# [  856.480923] floppy: error -5 while reading block 0
machine# [  856.489929] grub[19699]: segfault at 0 ip   (null) sp bfad0604 error 4 in grub[8048000+2b000]
machine# /nix/store/8mzqd12byvk327m5xmfy9kkzf9587cca-grub-0.97-patch-1.12/sbin/grub-install: line 448: 19699 Segmentation fault      $grub_shell --batch $no_floppy --device-map=$device_map  > $log_file <<EOF
machine# root $root_drive
machine# setup $force_lba --stage2=$grubdir/stage2 --prefix=$grub_prefix $install_drive
machine# quit
machine# EOF
```

Is that supposed to happen?

**Comment by vcunat** [comment]
In general, I don't know what to do about load/timing-sensitive tests, as they tend to generate lots of noise.

**Comment by domenkozar** [comment]
@exi that's an old thing that has been haunting us for years: https://github.com/NixOS/nixpkgs/issues/5640

**Comment by exi** [comment]
@vcunat timing issues still generally expose issues with the software under test or the test itself.

We probably need better ways to determine the state of the system before continuing with the test.

**Comment by domenkozar** [comment]
Maybe all timeouts should be overriden on Hydra, but then again the load could also happen on real server...

**Comment by exi** [comment]
@domenkozar @vcunat from my previous job where we had a lot of timing issues with end-to-end tests: 

every client->sleep(number of seconds) needs to go and be replaced with:
- wait for the next state where it is safe to continue by continually polling for "is X done yet?"
- then take the next step in the test.

Sadly, we have a lot of tests in nixos/tests that just sleep(5) and hope for the best.

**Comment by domenkozar** [comment]
@exi yes, but what do you do for systemd services that have the timeout set? Like how @shlevy described why mysql fails to start.

**Comment by vcunat** [comment]
Yeah, that's hard. I had mainly those in-test sleeps in mind. Maybe the systemd timeouts aren't that difficult to disable globally (for tests).

**Comment by wmertens** [comment]
The keyboard tests do a poll with a 1 minute timeout and they still fail 🙂

>

**Comment by exi** [comment]
@wmertens Not sure about the keyboard tests, but could it be possible that the system is under high load/swapping and 1 minute might just not be enough?
I would like to add timestamps to the log output to see how long it actually takes from keypress to its recognition.
In general, would it be possible to attach vnc or something to qemu and do a screen recording of the test? I would like to see a tab in hydra to show me a video of the failed test session.

**Comment by wmertens** [comment]
On Wed, Sep 14, 2016 at 7:49 AM Reno Reckling <notifications@github.com>
wrote:

> @wmertens <https://github.com/wmertens> Not sure about the keyboard
> tests, but could it be possible that the system is under high load/swapping
> and 1 minute might just not be enough?
> I would like to add timestamps to the log output to see how long it
> actually takes from keypress to its recognition.
>
Yes, timestamps would be great, I have a script that adds prefixes to
stdout and stderr but keeps them interleaved, should not be hard to also
add time deltas. Anyone opposed?


> In general, would it be possible to attach vnc or something to qemu and do
> a screen recording of the test? I would like to see a tab in hydra to show
> me a video of the failed test session.
>
¯\_(ツ)_/¯ dunno

**Comment by domenkozar** [comment]
We're down to 18 failures:

- pump.io will get removed unless someone updates it https://github.com/NixOS/nixpkgs/pull/18615
- mysqlWorkbench is being worked on by Karn
- x42-plugins fail to unpack?
- we should probably just disable the remaining kernel modules failures
- grsecurity, mathics and virtualbox tests need fixing
- it looks like @offlinehacker is busy, anyone else wants to upgrade `amdappsdkFull`?

**Comment by the-kenny** [comment]
I just fixed `x42-plugins` in master by removing `xz` from `buildInputs`. Strangely the fix doesn't work in `release-16.09`. Is there a difference in the stdenv that may cause this?

**Comment by domenkozar** [comment]
@the-kenny hm, transient error then?

grsecurity and virtualbox tests have both kernel panics: http://hydra.nixos.org/build/40684687 http://hydra.nixos.org/build/40690563

**Comment by vcunat** [comment]
The build has been randomly succeeding and failing on Hydra since at least 16.03. It was mentioned in this thread already.

**Comment by joachifm** [comment]
The grsec boot failure on i686 should be fixed with the latest version.

**Comment by domenkozar** [comment]
Here are the last items:

- amdappsdkFull
- grsecurity test
- virtualbox.net-hostonlyif

**Comment by rvl** [comment]
pump.io will pass if we merge #18574 and #18622.

**Comment by domenkozar** [comment]
Virtualbox is transient, grsecurity probably also due to the fact is succeeds on master: http://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.grsecurity.i686-linux

**Comment by domenkozar** [comment]
Two more failing, I've fixed them both. We might get a fully built evaluation for the first time in years!

Thank you everyone for participating! <3

**Comment by shlevy** [comment]
:astonished:

**Comment by magnetophon** [comment]
I'm the maintainer of x42-plugins, but I haven't been doing a great job:
For a while now, it has been alternating between building and failing with:
``do not know how to unpack source archive /...-x42-plugins-20160619.tar.xz``, see:
 https://hydra.nixos.org/build/41256475/nixlog/1#line-66

I have no idea what could cause this.

**Comment by vcunat** [comment]
> For a while now

I've seen the package do that months ago already (certainly somewhere in 16.03), but I failed to understand why (multiple comments above).

**Comment by domenkozar** [comment]
@vcunat could kde5 test failure be connected to mesa/wayland change? http://hydra.nixos.org/job/nixos/release-16.09/tested#tabs-constituents

**Comment by domenkozar** [comment]
@vcunat @ttuegel 4cf7839305df45d83b97fd14b27169b7d6a56ff7 breaks kde5 test, I just confirmed that.

Interesting part from the log:

```
machine# [  323.815824] xsession[875]: QMetaProperty::read: Unable to handle unregistered datatype 'KWayland::Server::SurfaceInterface*' for property 'KWin::Toplevel::surface'
machine# [  323.822767] xsession[875]: CreateNotify: 31457283
machine# [  323.885256] org.kde.kded5[918]: KCrash: Application 'kded5' crashing...
machine# [  323.887724] org.kde.kded5[918]: KCrash: Attempting to start /run/current-system/sw/lib/libexec/drkonqi from kdeinit
machine: must succeed: xwininfo -root -tree | sed 's/.*0x[0-9a-f]* \"\([^\"]*\)\".*/\1/; t; d'
machine: exit status 0
machine: no match 'kscreenlocker_greet' on '^Desktop '
machine: no match 'ksmserver' on '^Desktop '
machine: no match 'krunner' on '^Desktop '
machine: no match 'KWin' on '^Desktop '
machine# [  325.056211] xsession[875]: CreateNotify: 33554435
machine# [  326.043951] drkonqi[1897]: segfault at 0 ip           (null) sp 00007ffcbbecdb58 error 14 in drkonqi[400000+a3000]
machine: must succeed: xwininfo -root -tree | sed 's/.*0x[0-9a-f]* \"\([^\"]*\)\".*/\1/; t; d'
machine: exit status 0
machine: no match 'kscreenlocker_greet' on '^Desktop '
machine: no match 'ksmserver' on '^Desktop '
machine: no match 'krunner' on '^Desktop '
machine: no match 'KWin' on '^Desktop '
machine: must succeed: xwininfo -root -tree | sed 's/.*0x[0-9a-f]* \"\([^\"]*\)\".*/\1/; t; d'
machine: exit status 0
machine: no match 'kscreenlocker_greet' on '^Desktop '
machine: no match 'ksmserver' on '^Desktop '
machine: no match 'krunner' on '^Desktop '
machine: no match 'KWin' on '^Desktop '
machine# [  326.877214] xsession[875]: file:///nix/store/ims7bz253ajj693mz83y1yh8s4zvxbnq-plasma-workspace-5.7.4/share/plasma/look-and-feel/org.kde.breeze.desktop/contents/components/InfoPane.qml:54:22: Unable to assign [undefined] to int
machine# [  326.924325] xsession[875]: file:///nix/store/ims7bz253ajj693mz83y1yh8s4zvxbnq-plasma-workspace-5.7.4/share/plasma/look-and-feel/org.kde.breeze.desktop/contents/components/UserDelegate.qml:82:9: QML Image: Cannot open: file:///nix/store/ims7bz253ajj693mz83y1yh8s4zvxbnq-plasma-workspace-5.7.4/share/plasma/look-and-feel/org.kde.breeze.desktop/contents/components/user-identity
```

**Comment by ttuegel** [comment]
@domenkozar I can't reproduce this failure locally.

**Comment by domenkozar** [comment]
It's reverted now in 16.09
